From 91e4e59f0a15c0e0d938fb5fcb05e086ff73b08f Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Thu, 15 Oct 2015 17:38:29 +0800
Subject: [PATCH 0576/5242] MLK-11704 net: fec: fix MDIO read/write timeout
 when ethx down/up after suspend/resume

commit  d4aecb257f948bd5eb3c52790a77b1aac7daf0b3 from
https://source.codeaurora.org/external/imx/linux-imx.git

This reverts commit 12fee3d24574bd084e8b6fa1dcaf84a084298c9a.

The commit:12fee3d24574 intruduce MDIO read/write timeout when Ethernet
down and up after Mega/Fast off. The patch just revert the patch.

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/freescale/fec_main.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index d2848b1..94d2e52 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -3001,6 +3001,7 @@ static inline bool fec_enet_irq_workaround(struct fec_enet_private *fep)
 	}
 
 	phy_disconnect(ndev->phydev);
+	ndev->phydev = NULL;
 
 	if (fep->quirks & FEC_QUIRK_ERR006687)
 		imx6q_cpuidle_fec_irqs_unused();
-- 
1.7.9.5

